### PR TITLE
Allow building with GHC 9.0

### PR DESCRIPTION
**Pull request by RyanGlScott**
~~There are still some downstream dependency issues involving `th-abstraction` (most likely bos/aeson#809) that prevent a build plan from being discovered.~~ **EDIT**: All fixed now!

### CODE REVIEW

File: src/Control/Lens/Prism.hs [hunk 1]
```diff
@@ -132,13 +132,13 @@ outside k = withPrism k $ \bt seta f ft ->
...
         -> Prism (Either s u) (Either t v) (Either a c) (Either b d)
-without k =
-  withPrism k         $ \bt seta k' ->
-  withPrism k'        $ \dv uevc    ->
-  prism (bimap bt dv) $ \su ->
-  case su of
-    Left s  -> bimap Left Left (seta s)
-    Right u -> bimap Right Right (uevc u)
+without k uvcd =
+  (withPrism k         $ \bt seta k' ->
```
Comment (RyanGlScott) [inline src/Control/Lens/Prism.hs:136]: Can anyone think of a cleaner way to accomplish this?

Comment (phadej) [reply src/Control/Lens/Prism.hs:136]: Manually inlining `withPrism` definition...

Comment (phadej) [reply src/Control/Lens/Prism.hs:136]: Or just

```
without :: APrism s t a b
        -> APrism u v c d
        -> Prism (Either s u) (Either t v) (Either a c) (Either b d)
without k k' =
   withPrism k         $ \bt seta ->
   withPrism k'        $ \dv uevc    ->
   prism (bimap bt dv) $ \su ->
   case su of
     Left s  -> bimap Left Left (seta s)
     Right u -> bimap Right Right (uevc u)
```

Comment (RyanGlScott) [reply src/Control/Lens/Prism.hs]: https://github.com/ekmett/lens/pull/942#discussion_r498707113 is a fantastic idea. Suggestion applied.

**Pull request merged**